api/v1alpha1/oadp_types.go

```diff
@@ -130,26 +130,53 @@ type PodConfig struct {
 	Env []corev1.EnvVar `json:"env,omitempty"`
 }
 
-// ResticConfig is the configuration for restic server
-type ResticConfig struct {
+type NodeAgentCommonFields struct {
 	// enable defines a boolean pointer whether we want the daemonset to
 	// exist or not
 	// +optional
 	Enable *bool `json:"enable,omitempty"`
-	// supplementalGroups defines the linux groups to be applied to the Restic Pod
+	// supplementalGroups defines the linux groups to be applied to the NodeAgent Pod
 	// +optional
 	SupplementalGroups []int64 `json:"supplementalGroups,omitempty"`
-	// timeout defines the Restic timeout, default value is 1h
+	// timeout defines the NodeAgent timeout, default value is 1h
 	// +optional
 	Timeout string `json:"timeout,omitempty"`
 	// Pod specific configuration
 	PodConfig *PodConfig `json:"podConfig,omitempty"`
 }
 
+// NodeAgentConfig is the configuration for node server
+type NodeAgentConfig struct {
+	// Embedding NodeAgentCommonFields
+	// +optional
+	NodeAgentCommonFields `json:",inline"`
+
+	// The type of uploader to transfer the data of pod volumes, the supported values are 'restic' or 'kopia'
+	// +kubebuilder:validation:Enum=restic;kopia
+	// +kubebuilder:validation:Required
+	UploaderType string `json:"uploaderType"`
+}
+
+// ResticConfig is the configuration for restic server
+type ResticConfig struct {
+	// Embedding NodeAgentCommonFields
+	// +optional
+	NodeAgentCommonFields `json:",inline"`
+}
+
 // ApplicationConfig defines the configuration for the Data Protection Application
 type ApplicationConfig struct {
 	Velero *VeleroConfig `json:"velero,omitempty"`
+	// (deprecation warning) ResticConfig is the configuration for restic server.
+	// Restic is for backwards compatibility and is replaced by the nodeAgentConfig
+	// Restic will be removed with the OADP 1.4
+	// +kubebuilder:deprecatedversion:warning=1.3
+	// +optional
 	Restic *ResticConfig `json:"restic,omitempty"`
+
+	// NodeAgent is needed to allow selection between kopia or restic
+	// +optional
+	NodeAgent *NodeAgentConfig `json:"nodeAgentConfig,omitempty"`
```

Comment:
@weshayutin @sseago 

I know the design says 'nodeAgentConfig', but should we actually at this point name it 'nodeAgent' ?

Why? When I am playing with this, it seems for me more natural to write it without `Config` suffix as it's already under `configuration` section in our yaml/definition.:

```yaml
  configuration:
    nodeAgent:
      enable: true
      uploaderType: restic
```

Then:
```yaml
  configuration:
    nodeAgentConfig:
      enable: true
      uploaderType: restic
```

Reply:
Note that the yaml will still have `configuration.nodeAgent` since the *field name* is `NodeAgent`, while the field type is `NodeAgentConfig`.

That said, I don't oppose renaming the struct to be the same as the field name where it's used. I think we had `Config` following prior practice with `ResticConfig`, but having struct and field named `NodeAgent` is fine by me. Just update the design in this PR along with the impl change so we don't have inconsistent docs.

Reply:
I misread this. The field is indeed called `NodeAgent`, but the yaml says `nodeAgentConfig` -- so at a minimum, change yaml to `nodeAgent` -- I don't have a strong opinion on whether the struct itself changes name.

Reply:
I have changed the config option to `nodeAgent` and kept the structure name `NodeAgentConfig` so it's similar to `restic` option and `ResticConfig` structure name.